app/controllers/unsubscriptions_controller.rb

```diff
@@ -28,13 +36,34 @@ def confirmed
 
   def set_attributes
     @id = params.require(:id)
-    @subscription = api.get_subscription(@id)
-    @title = @subscription.dig("subscription", "subscriber_list", "title").presence
-    @authenticated_for_subscription = check_authenticated(@subscription)
+    @original_subscription = api.get_subscription(@id)
+    @latest_subscription = api.get_latest_matching_subscription(@id)
+    @title = @latest_subscription.dig("subscription", "subscriber_list", "title").presence
+    @authenticated_for_subscription = check_authenticated(@latest_subscription)
+  end
+
+  def subscription_is_latest?
+    id(@original_subscription) == id(@latest_subscription)
+  end
+
+  def old_subscription_has_same_frequency_as_latest?
+    frequency(@latest_subscription) == frequency(@original_subscription)
+  end
+
+  def subscription_ended?(subscription)
+    subscription.dig("subscription", "ended_at").present?
   end
 
-  def subscription_ended?
-    @subscription.dig("subscription", "ended_at").present?
+  def id(subscription)
+    subscription.dig("subscription", "id")
+  end
+
+  def frequency(subscription)
+    subscription.dig("subscription", "frequency")
+  end
+
+  def already_unsubscribed
```

Comment:
I think this method should also have a question mark at the end of the name

Reply:
Actually it can be removed altogether! Good spot.